### DISCUSSION TITLE
Java+Kotlin

### DISCUSSION THREAD

**Discussion by itamarsch** [BrenekH/kotlin-for-frc#118] [Q&A] [answered]
Is it possible to import kotlin code into the java files?
and if not is there a way to set up the project for kotlin and java use? (simultaneously)

**Comment by BrenekH** [comment]
I'm not sure what you mean by importing Kotlin code, but you can [call Kotlin from Java (Jetbrains Docs)](https://kotlinlang.org/docs/java-to-kotlin-interop.html).

I'm sure there are also plenty of other resources out there about using Java and Kotlin in the same project.

**Comment by keithchad** [comment]
> Is it possible to import kotlin code into the java files? and if not is there a way to set up the project for kotlin and java use? (simultaneously)

Yes. You can use both Kotlin and Java code in your project file due to interoperability which enables the use of Kotlin code in Java and Java code in Kotlin. Just create your Java file and Kotlin file and access the functions in either Java or Kotlin by calling them the way you usually do.

**Comment by itamarsch** [comment, marked as answer]
after searching a bit I found out this is a vscode problem with an open isue since 2018 https://github.com/redhat-developer/vscode-java/issues/531
